string name
float32[] activations